crates/types/src/data.rs

```diff
@@ -611,10 +612,16 @@ pub struct SequencingLeaf<TYPES: NodeType> {
     /// So we can ask if it extends
     pub parent_commitment: Commitment<Self>,
 
-    /// Either the block number and payload, or the block header.
-    pub deltas: Either<(u64, TYPES::BlockPayload), TYPES::BlockHeader>,
+    /// Block header.
+    pub block_header: TYPES::BlockHeader,
 
-    /// Transactions that were marked for rejection while collecting deltas
+    /// Set of commitments to the contained transactions.
+    ///
+    /// It may be empty for nodes not in the DA committee.
+    pub transaction_commitments:
+        HashSet<Commitment<<TYPES::BlockPayload as BlockPayload>::Transaction>>,
```

Comment:
I think `Option<Vec<_>>` would be a better data structure here. `Option` to make the "may be empty" case more explicit and `Vec` because we very much care about the order of the transactions.

Also, why are we using transaction commitments? We need the application to be able to recover the entire block contents somehow, which is why I kind of feel we should just have an `Option<BlockPayload>`, at least by the time this gets put into a `Decide` event

Reply:
Agree! I'm going to change it to `Option<BlockPayload>`.